agent/functions/put_file_test.go

```diff
@@ -0,0 +1,45 @@
+package functions_test
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github/clover0/github-issue-agent/functions"
+)
+
+func TestPutFile(t *testing.T) {
+	t.Run("successfully writes content to file", func(t *testing.T) {
```

Comment:
from CodeQualityAnalyst
テストデータ用のディレクトリが存在しない場合に備えて、ディレクトリを作成するコードを追加すると良いでしょう。

```suggestion
os.MkdirAll("testdata", os.ModePerm)
```